lib/parsers/yarn2-lock-parser.ts

```diff
@@ -81,21 +93,57 @@ export class Yarn2LockParser extends LockParserBase {
     return depTreeWithMeta;
   }
 
-  protected getDepMap(lockfile: Lockfile): DepMap {
+  protected getDepMap(
+    lockfile: Lockfile,
+    resolutions?: ManifestDependencies,
+  ): DepMap {
     const yarnLockfile = lockfile as Yarn2Lock;
     const depMap: DepMap = {};
 
+    const dependencies = lockfile.dependencies || {};
+    const resolutionPaths = resolutions ? Object.keys(resolutions) : [];
+
     for (const [depName, dep] of Object.entries(yarnLockfile.object)) {
+      const depNameNoVer = depName
```

Comment:
You can find a better way to do this :)
The only time you'll have more than one @ character is when the package is scoped (I.e. @snyk/something)
You can easily solve it with regex, something like '/.*@.*/' should work